migration-tests/pallets/indices.ts

```diff
@@ -0,0 +1,86 @@
+import '@polkadot/api-augment';
+import { PalletTest } from '../types.js';
+import assert from 'assert';
+
+const formatIndicesEntry = ([key, value]: any) => {
+    const index = key.args[0].toNumber();
+    if (value.isSome) {
+        const [who, deposit, frozen] = value.unwrap();
+        return {
+            index,
+            who: who.toString(),
+            deposit: deposit.toBigInt(),
+            frozen: frozen.isTrue
+        };
+    }
+    return null;
+};
+
+export const indicesTests: PalletTest = {
+    pallet_name: 'indices',
+    pre_check: async ({ ah_api_before }) => {
+        const indices = await ah_api_before.query.indices.accounts.entries();
+        assert(indices.length === 0, "[AH] Assert storage indices.accounts() is empty before migration");
+    },
+    post_check: async ({ rc_api_before, rc_api_after, ah_api_before, ah_api_after }) => {
+        const indices = await rc_api_after.query.indices.accounts.entries();
+        assert(indices.length === 0, "[RC] Assert storage indices.accounts() is empty after migration");
+
+        // Get RC indices before migration
+        const rc_indices_before = await rc_api_before.query.indices.accounts.entries();
+        const rc_before_formatted = rc_indices_before
+            .map(formatIndicesEntry)
+            .filter((entry): entry is NonNullable<typeof entry> => entry !== null);
+
+        // Get AH indices before migration
+        const ah_indices_before = await ah_api_before.query.indices.accounts.entries();
+        const ah_before_formatted = ah_indices_before
+            .map(formatIndicesEntry)
+            .filter((entry): entry is NonNullable<typeof entry> => entry !== null);
+
+        // Convert to Map for comparison (equivalent to BTreeMap in Rust)
+        const all_pre = new Map(
+            [...rc_before_formatted, ...ah_before_formatted].map(({ index, who, deposit, frozen }) => 
+                [index, { who, deposit, frozen }]
+            )
+        );
+
+        // Assert that we have indices to migrate
+        assert(all_pre.size > 0, "[AH] Assert storage indices.accounts() is not empty before migration");
```

Comment:
Going forward, use `expect(...).toBeGreaterThan` over raw `assert`s; in general, really.

In TS, `assert`s just fail and tell you no information why, whereas `expect` will show you a diff.